src/docs/development/ui/actions_and_shortcuts.md

```diff
@@ -0,0 +1,595 @@
+handler closure if the intent has a mapping to an enabled action, and returns
+null if it doesn't, so that the button is disabled if there is no matching
+enabled action in the context:
+
+```dart
+@override
+Widget build(BuildContext context) {
+  return Actions(
+    actions: <Type, Action<Intent>>{
+      SelectAllIntent: SelectAllAction(),
+    },
+    child: Builder(
+      builder: (BuildContext context) => TextButton(
+        child: const Text('SELECT ALL'),
+        onPressed: Actions.handler<SelectAllIntent>(
+          context,
+          SelectAllIntent(controller: controller),
+        ),
+      ),
+    ),
+  );
+}
+```
+
+The `Actions` widget only invokes actions when `isEnabled(Intent intent)`
+returns true, allowing the action to decide if the dispatcher should consider it
+for invocation.  If the action isn't enabled, then the `Actions` widget gives
+another enabled action higher in the widget hierarchy (if it exists) a chance to
+execute.
+
+The previous example uses a `Builder` because `Actions.handler` and
+`Actions.invoke` (for example) only finds actions in the provided `context`, and
+if the example passes the `context` given to the `build` function, the framework
+starts looking _above_ the current widget.  Using a `Builder` allows the
+framework to find the actions defined in the same `build` function.
+
+It is possible to invoke an action without needing a `BuildContext`, but since
+the `Actions` widget requires a context to find an enabled action to invoke, you
+need to provide one, either by creating your own `Action` instance, or by
+finding one in an appropriate context with `Actions.find`.
+
+To invoke the action, pass the action to the `invoke` method on an
+`ActionDispatcher`, either one you created yourself, or one retrieved from an
+existing `Actions` widget using the `Actions.of(context)` method. Check whether
+the action is enabled before calling `invoke`. Of course, you can also just call
+`invoke` on the action itself, passing an `Intent`, but then you opt out of any
+services that an action dispatcher might provide (like logging, undo/redo, and
+so on).
+
+### Action dispatchers
+
+Most of the time, you just want to invoke an action, have it do its thing, and
+forget about it. Sometimes, however, you might want to log the executed actions.
+
+This is where replacing the default `ActionDispatcher` with a custom dispatcher
+comes in.  You pass your `ActionDispatcher` to the `Actions` widget, and it
+invokes actions from any `Actions` widgets below that one that doesn't set a
+dispatcher of its own.
+
+The first thing `Actions` does when invoking an action is look up the
+`ActionDispatcher` and pass the action to it for invocation. If there is none,
+it creates a default `ActionDispatcher` that simply invokes the action.
+
+If you want a log of all the actions invoked, however, you can create your own
+`LoggingActionDispatcher` to do the job:
+
```

Comment:
Well, it would compile, but it wouldn't do anything. I don't think it makes sense in Dartpad.  The final "putting it together" section now has a functional app with this code in it.